### PR TITLE
Use @nextcloud/sharing in files and files_sharing

### LINKED ISSUE #28488: Replace files_sharing constants with @nextcloud/sharing types
https://github.com/nextcloud/server/blob/0244368e3598245c077714ca2a6bb1843d416cbf/apps/files_sharing/src/index.js#L25-L38 registers global variables for the sharing type constants. These are now also in https://www.npmjs.com/package/@nextcloud/sharing.

Let's use the common constants instead of globals.

The globals are used in the `files` and `files_sharing` apps.

cc @artonge @Pytal for a little clean-up task :)

### PR DESCRIPTION
**Pull request by artonge**
Use the new `@nextcloud/sharing` package for sharing constants.

There are still some references to `OC.Share` in legacy js files, but they can't be removed as they do not support module imports.

Fix https://github.com/nextcloud/server/issues/28488

### CODE REVIEW

File: apps/files_sharing/src/share.js [hunk 1]
```diff
@@ -70,7 +72,7 @@ import escapeHTML from 'escape-html'
 		 */
 		attach: function(fileList) {
 			// core sharing is disabled/not loaded
-			if (!OC.Share) {
+			if (!ShareTypes) {
```
[the remark this answers is not in the document]
Comment (artonge) [reply apps/files_sharing/src/share.js:75]: - method on SharingManager

File: apps/files_sharing/src/mixins/SharesMixin.js [hunk 2]
```diff
@@ -74,18 +74,6 @@ export default {
...
-				SHARE_TYPE_USER: OC.Share.SHARE_TYPE_USER,
-				SHARE_TYPE_GROUP: OC.Share.SHARE_TYPE_GROUP,
-				SHARE_TYPE_LINK: OC.Share.SHARE_TYPE_LINK,
-				SHARE_TYPE_EMAIL: OC.Share.SHARE_TYPE_EMAIL,
-				SHARE_TYPE_REMOTE: OC.Share.SHARE_TYPE_REMOTE,
-				SHARE_TYPE_CIRCLE: OC.Share.SHARE_TYPE_CIRCLE,
-				SHARE_TYPE_GUEST: OC.Share.SHARE_TYPE_GUEST,
-				SHARE_TYPE_REMOTE_GROUP: OC.Share.SHARE_TYPE_REMOTE_GROUP,
-				SHARE_TYPE_ROOM: OC.Share.SHARE_TYPE_ROOM,
-			},
```
Comment (MichaIng) [inline apps/files_sharing/src/mixins/SharesMixin.js:88]: Is the replacement missing here or is it not required (anymore)?

**Review by Shar1988 — COMMENTED** [review]
const get = (from, ...selectors) =>
  [...selectors].map(s =>
    s
      .replace(/\[([^\[\]]*)\]/g, '.$1.')
      .split('.')
      .filter(t => t !== '')
      .reduce((prev, cur) => prev && prev[cur], from)
  );

**Comment by MichaIng** [comment]
Is it possible to import the constants themselves instead of the array? Their unique prefixes make a name spacing doubled, so this would make code look a little cleaner IMHO 🙂:
```js
if (share.type === ShareTypes.SHARE_TYPE_GROUP) {
```
=>
```js
if (share.type === SHARE_TYPE_GROUP) {
```
But if this means that each type needs to be imported individually, then the import block came much larger instead 😄.

File: apps/files_sharing/src/mixins/SharesMixin.js (hunk 2 above)
Comment (artonge) [reply apps/files_sharing/src/mixins/SharesMixin.js:88]: It was actually dead code before the PR, I think, as `ShareTypes` was already included as a mixin a few line above: https://github.com/nextcloud/server/pull/28561/files#diff-68f3cd4065b52c17ac4302563e4cf7eaced8e7a5d93a99e54f81ca090fa4f9adL38

**Comment by artonge** [comment]
> Is it possible to import the constants themselves instead of the array? Their unique prefixes make a name spacing doubled, so this would make code look a little cleaner IMHO slightly_smiling_face:
> 
> ```js
> if (share.type === ShareTypes.SHARE_TYPE_GROUP) {
> ```
> 
> =>
> 
> ```js
> if (share.type === SHARE_TYPE_GROUP) {
> ```
> 
> But if this means that each type needs to be imported individually, then the import block came much larger instead smile.

I agree that the current state of the code is a bit repetitive, but as you said, it is cleaner in the import section, so I think it is better to keep the extra namespace. Also, I like it when constants are bond to an object, it is easier to understand where they are coming from.

**Comment by MichaIng** [comment]
> it is easier to understand where they are coming from

I agree with that. Also now one can easily find all uses of that module by searching all cases of "ShareTypes". So 2:1 for keeping it like it is 🙂.

**Review by skjnldsv — APPROVED** [review]

File: apps/files_sharing/src/share.js (hunk 1 above)
[the remark this answers is not in the document]
Comment (skjnldsv) [reply apps/files_sharing/src/share.js:75]: With the capabilities
```suggestion
			if (!getCapabilities().files_sharing?.api_enabled) {
```

**Comment by artonge** [comment]
Drone CI failing:
- samba: unrelated, failing in master too

**Review by MichaIng — APPROVED** [review]

**Pull request merged**